agent/consul/state/intention.go

```diff
@@ -70,6 +74,10 @@ func intentionsTableSchema() *memdb.TableSchema {
 				Unique:       true,
 				Indexer: &memdb.CompoundIndex{
 					Indexes: []memdb.Indexer{
+						&memdb.StringFieldIndex{
```

Comment:
This is needed because this table is used for duplicate checking. Currently it's not possible to have duplicates between each source type because `consul` doesn't allow a `spiffe://` prefix and `external-trust-domain` doesn't allow a path and `external-uri` requires a path. However it's best not to rely on this.